File: COMP9041/lab06/echon.py
#!/usr/bin/env python

# repeat echo in python

import sys,re

if len(sys.argv) != 3: # echon.py is included
    # sys.stderr.write("Usage: ./echon.py <number of lines> <string>")
    # print sys.stderr
    print "Usage: ./echon.py <number of lines> <string>"
    exit(1)

pattern = "^[0-9]+$"

if not re.match(pattern, sys.argv[1]):
    # sys.stderr.write("./echon.py: argument 1 must be a non-negative integer")
    print "./echon.py: argument 1 must be a non-negative integer"
    exit(1)
    
for i in range(int(sys.argv[1])):
    print sys.argv[2]

File: COMP9041/lab05/echon.pl
#!/usr/bin/perl -w

# 2 arguments, if not 2 --> STDERR
# if the first arguments is not non-negative integer --> STDERR

# Check the number of argv
$num_argv = @ARGV;
if($num_argv != 2){
    print STDERR "Usage: ./echon.pl <number of lines> <string>\n";
    exit(1);
}

# Check whether the first argv is a non-negative integer
$repeat = $ARGV[0];
if($repeat =~ m/\D/){
    print STDERR "./echon.pl: argument 1 must be a non-negative integer\n";
    exit(1);
}
if($repeat < 0){
    print STDERR "./echon.pl: argument 1 must be a non-negative integer\n";
    exit(1);
}

# loop
$count = 0;
while($count < $repeat){
    print "$ARGV[1]\n";
    $count++;
}
